packages/react-kit/src/components/modal/components/Redeem/RedeemNonModal.tsx

```diff
@@ -41,15 +42,132 @@ import { useConvertionRate } from "../../../widgets/finance/convertion-rate/useC
 import NonModal, { NonModalProps } from "../../nonModal/NonModal";
 import { useConfigContext } from "../../../config/ConfigContext";
 import {
+  RedemptionContextProps,
   RedemptionWidgetAction,
   useRedemptionContext
 } from "../../../widgets/redemption/provider/RedemptionContext";
 import { BosonFooter } from "./BosonFooter";
 import { theme } from "../../../../theme";
 import { useAccount } from "../../../../hooks/connection/connection";
 import { RedeemHeader } from "./RedeemHeader";
+import { isTruthy } from "../../../../types/helpers";
+import { useSellers } from "../../../../hooks/useSellers";
+import { ethers } from "ethers";
+import { subgraph } from "@bosonprotocol/core-sdk";
+import styled from "styled-components";
 
 const colors = theme.colors.light;
+const UlWithWordBreak = styled.ul`
+  * > {
+    word-break: break-word;
+  }
+`;
+const checkSignatures = ({
+  doFetchSellersFromSellerIds,
+  parentOrigin,
+  sellerIds,
+  signatures,
+  sellersFromSellerIds,
+  areSignaturesMandatory
+}: Pick<RedeemNonModalProps, "parentOrigin" | "sellerIds" | "signatures"> & {
+  doFetchSellersFromSellerIds: boolean;
+  sellersFromSellerIds: subgraph.SellerFieldsFragment[] | undefined;
+  areSignaturesMandatory: boolean;
+}) => {
+  try {
+    if (areSignaturesMandatory && !sellerIds) {
+      return (
+        <p>
+          SellerIds must be defined as these are defined postDeliveryInfoUrl,
+          deliveryInfoHandler, redemptionSubmittedHandler,
+          redemptionConfirmedHandler{" "}
+        </p>
+      );
+    }
+    if (
+      sellerIds?.length &&
+      areSignaturesMandatory &&
+      (!signatures || signatures?.filter(isTruthy).length !== sellerIds.length)
+    ) {
+      return (
+        <p>
+          Please provide a list of signatures of the message{" "}
+          {JSON.stringify({ origin: "<parentWindowOrigin>" })} for each seller
+          in sellerIds list
+        </p>
+      );
+    }
+    if (doFetchSellersFromSellerIds && !sellersFromSellerIds) {
+      return (
+        <p>
+          Could not retrieve sellers from the specified sellerIds {sellerIds}
+        </p>
+      );
+    }
+    const originMessage = JSON.stringify({ origin: parentOrigin });
+    const firstIndexSignatureThatDoesntMatch = sellersFromSellerIds?.findIndex(
+      ({ admin }, index) => {
+        if (!signatures?.[index]) {
+          return true;
+        }
+        const signerAddr = ethers.utils.verifyMessage(
+          originMessage,
+          signatures[index]
+        );
+        if (signerAddr.toLowerCase() !== admin.toLowerCase()) {
```

Comment:
in case the seller is using LENS (seller.authTokenType === 2), the admin is 0x0000000000000000000000000000000000000000
In this case, need to look at the owner of the seller.authTokenId...
(https://github.com/bosonprotocol/core-components/blob/fix-vulnerability-signature/packages/core-sdk/src/erc721/mixin.ts#L14 with the LENS contract address)